sdk/docs/iot/README.md

```diff
@@ -1,81 +1,60 @@
-# Azure IoT Clients
+# Azure Embedded C SDK IoT Clients
```

Comment:
We intentionally removed `Embedded C SDK` here before so that it doesn't repeatedly show up in this list:
https://azuresdkdocs.blob.core.windows.net/$web/c/docs/1.0.0/pages.html
![image](https://user-images.githubusercontent.com/6527137/94755156-614e9280-0348-11eb-9f8a-c8d51dde3448.png)

See https://github.com/Azure/azure-sdk-for-c/pull/741
https://github.com/Azure/azure-sdk-for-c/pull/735#issuecomment-627107333

Reply:
Thank you!  I was only 8 days old at this point in time 😄